frontend/src/components/pages/profile/ProfileCard/variants/OrgCard/OrgAction.tsx

```diff
@@ -0,0 +1,87 @@
+import { Organization } from "@interfaces/organizations";
+import { Typography, Box, Menu, CardActions, MenuItem, CardActionArea } from "@material-ui/core";
+import OrgListItem from "./OrgListItem";
+import { useState } from "react";
+import Link from "next/link";
+import { ProfileActionProps } from "@components/pages/profile/ProfileCard/variants/ProfileCardBase";
+import { useRouter } from "next/router";
+
+const orgLink = (orgId: string) => `/orgs/${orgId}`;
+
+export type Props = {
+  orgs: Pick<Organization, "id" | "name" | "logoUrl">[];
+};
+
+/**
+ * Function that returns a functional component for use as a profile card action.
+ * The outer function takes the props not required by the card action.
+ * The inner function takes the remaining props as required for the card.
+ *
+ * This function takes a list of orgs, and returns different components depending on its length:
+ * - No orgs: Text
+ * - 1 org: List to its org admin page
+ * - Multiple orgs: Menu with links to their admin pages
+ */
+export const createOrgProfileAction = ({ orgs }: Props): React.VFC<ProfileActionProps> => {
```

Comment:
I have some concerns regarding this pattern that I need some time to write some thoughts about.

Reply:
I agree that this could ideally be done in a cleaner way. However, I couldn't find another way to pass parts of the props to a component in one place, and then the rest in another place. From what I could find, it seems that _component instantiation_ can only happen in one place. Would love feedback on possible alternative ways to do this!

The basic pattern here is just that of wrapping the inner function in a closure to pass the partial arguments. I think this is a fine pattern in functional JavaScript, as long as it is well documented (as I tried to do). I am curious if this has a negative impact on React's rendering though.

Reply:
I'll try to summarize my thoughts here:
* In general, I am apprehensive about functions that return a functional component, and I struggle to see why they should not be refactored into components instead. I.e., why is this not just a functional component that accepts `Organization[]` as props, and rather than passing a `createProfileAction` to the cards as a prop, pass the component itself.
* The refactor into `createProfileAction` and `createOrgProfileAction` introduces significant complexity to the component design. In my opinion, such significant deviation from the original component design should likely be resolved by accepting an optional child component for the profile card that could render instead of the existing link component.
* I am unsure about the rendering implications of this, and it might be necessary to wrap the function call in a useCallback. However, this concern is alleviated with the suggestions above.

Reply:
I'll create a quick PR to demonstrate my suggestion

Reply:
Great contribution! Merged it in now.

I think I got lost in where the props were applied, and didn't think that the `data-test-id` could just as well be applied outside of `ProfileCardBase`.

Still, as an academic exercise, I am curious what the best solution is when you _do_ need to apply partial props in one place, and the rest in another. Are higher-order functions the way to go, or some other solution? Or should you always be able to refactor yourself out of such a situation?